utils/s2n_safety.h

```diff
@@ -91,7 +91,7 @@ static inline void* trace_memcpy_check(void *restrict to, const void *restrict f
   } while (0)
 
 /* Check for specific classes of errors */
-#define ERR_IS_BLOCKING( x )    ( x == S2N_ERR_BLOCKED || x == S2N_CALLBACK_BLOCKED )
+#define ERR_IS_BLOCKING( x )    ( (x) == S2N_ERR_BLOCKED || (x) == S2N_ERR_APPLICATION_BLOCKED )
```

Comment:
The original intent here was to allow customers to return a contractual error code indicating they were blocked. Removing S2N_CALLBACK_BLOCKED breaks that contract. I would assume that the async test would fail here, since it hasn't been updated and returns -2.

Reply:
GUARD_NONBLOCKING is also changed correspondingly to avoid failure here